okhttp/src/main/java/okhttp3/internal/Util.kt

```diff
@@ -171,6 +171,17 @@ fun Array<String>.concat(value: String): Array<String> {
   return result as Array<String>
 }
 
+/**
+ * Removes the first element of this non-empty set and returns it. If this set was created with
+ * [mutableSetOf] this returns the oldest element in the set.
+ */
+fun <T : Any?> MutableSet<T>.removeFirst(): T {
```

Comment:
Probably worth apply this to LinkedHashSet instead of MutableSet, since MutableSet doesn't have the ordering guarantees.

```
 * A generic unordered collection of elements that does not support duplicate elements, and supports
 * adding and removing elements.
```